### ISSUE TITLE
lr_schedule is wrong and gs_conf is non-convergent

### ISSUE THREAD

**Issue by zhshe** [xiaomi-research/dggt#49]
1. 
train.py
step in LambdaLR is epoch
but scheduler.step() be called in each batch

```
    warmup_iterations = 1000
    scheduler = LambdaLR(
        optimizer,
        lr_lambda=lambda step: min((step + 1) / warmup_iterations, 1.0) * 0.5 * (
            1 + torch.cos(torch.tensor(torch.pi * step / args.max_epoch)))
    )
```

```
for step in tqdm(range(args.max_epoch)):
    for batch in dataloader:
        with torch.cuda.amp.autocast(dtype=dtype):
            ...
            # #calculate loss
            # semantic_loss = semantic_loss_fn(semantic_logits[0].permut
            # loss = loss + 0.01 * semantic_loss

            if step >= 0:
                lpips_val = lpips_loss_fn(rendered_image, target_image)
                loss += 0.05 * min(step / 1000, 1.0) * lpips_val.mean() # *

        loss.backward()
        optimizer.step()
        scheduler.step()

```

2. why gs_conf is always non-convergent，but total loss is descent，is it correct?

<img width="1284" height="660" alt="Image" src="https://github.com/user-attachments/assets/7a036293-7839-405c-a9f8-b27585aa10cf" />

**Comment by Ziyi-Xiong1** [comment]
Thanks for pointing this out. This is indeed a code bug, and we will update the repository to fix it soon.
